cranelift/codegen/meta/src/gen_types.rs

```diff
@@ -48,6 +48,32 @@ fn emit_dynamic_vectors(bits: u64, fmt: &mut srcgen::Formatter) {
 
 /// Emit types using the given formatter object.
 fn emit_types(fmt: &mut srcgen::Formatter) {
+    fmt.line("#[allow(dead_code)]");
+    fmt.line("mod constants {");
+    fmt.indent(|fmt| {
+        fmtln!(
+            fmt,
+            "pub const LANE_BASE: u16 = {:#X};\n",
+            constants::LANE_BASE
+        );
+        fmtln!(
+            fmt,
+            "pub const REFERENCE_BASE: u16 = {:#X};\n",
+            constants::REFERENCE_BASE
+        );
+        fmtln!(
+            fmt,
+            "pub const VECTOR_BASE: u16 = {:#X};\n",
+            constants::VECTOR_BASE
+        );
+        fmtln!(
+            fmt,
+            "pub const DYNAMIC_VECTOR_BASE: u16 = {:#X};\n",
+            constants::DYNAMIC_VECTOR_BASE
+        );
```

Comment:
I don't feel strongly about this, but I'd be tempted to use a macro here to ensure that the name of the constant is always the same in the generated source as it is in the meta crate. Something like `constants!(fmt, LANE_BASE, REFERENCE_BASE, ...)` I guess.

I'm happy to merge this as-is, but I thought I'd mention it since I think `publish.rs` needs to be fixed anyway. Let me know whether you want to change this too.